### PR TITLE
Fixed #24963 -- Adds seekable property to files in python3

### PR DESCRIPTION
**Pull request by carsongee**

### CODE REVIEW

File: django/core/files/utils.py [hunk 1]
```diff
@@ -18,6 +18,7 @@ def __init__(self, file):
     readline = property(lambda self: self.file.readline)
     readlines = property(lambda self: self.file.readlines)
     seek = property(lambda self: self.file.seek)
+    seekable = property(lambda self: self.file.seekable)
```
Comment (MoritzS) [inline django/core/files/utils.py]: Would
```python
if six.PY2:
    seekable = property(lambda self: self.file.seekable)
```
be better here?

Comment (carsongee) [inline django/core/files/utils.py]: I can be explicit about it adding the property if that is preferred.  That was my original intention, but my test showed that the attribute wasn't added when you are in py2 so it seemed cleaner to leave it that way.  Let me know your preference, and I'll change it if requested

Comment (MoritzS) [inline django/core/files/utils.py]: It actually *does* get added, it's just that `lamdba self: self.file.seekable` raises an AttributeError and therefore `hasattr(file, 'seekable')` returns false in py2.

Comment (carsongee) [inline django/core/files/utils.py]: A subtle but important distinction, thanks for pointing that out.  I've updated it to add the version check.

File: tests/file_storage/tests.py [hunk 2]
```diff
@@ -403,6 +403,19 @@ def test_delete_no_name(self):
         with self.assertRaises(AssertionError):
             self.storage.delete('')
 
+    def test_seekable_(self):
```
Comment (MoritzS) [inline tests/file_storage/tests.py]: Just nitpicking, why the trailing underscore here?

Comment (carsongee) [inline tests/file_storage/tests.py]: whoops, nice catch.  Fixed.

**Comment by carsongee** [comment]
I saw your commend on the ticket and updated the docs.  Thanks for pointing me to that, I wasn't sure where to update it.

File: docs/ref/files/file.txt [hunk 3]
```diff
@@ -89,7 +89,8 @@ The ``File`` Class
     the following attributes and methods of its ``file`` object:
     ``encoding``, ``fileno``, ``flush``, ``isatty``, ``newlines``,
     ``read``, ``readinto``, ``readlines``, ``seek``, ``softspace``, ``tell``,
-    ``truncate``, ``writelines``, ``xreadlines``.
+    ``truncate``, ``writelines``, ``xreadlines``.  If you are using
+    python 3.x, it also adds the ``seekable`` method.
```
Comment (MoritzS) [inline docs/ref/files/file.txt]: In the docs it's generally written as "Python 3".

**Comment by MoritzS** [comment]
Maybe use `.. versionchanged:: 1.9` or add an entry to the release notes (`docs/releases/1.9.txt`) or both to make it clear were it got changed.

**Comment by carsongee** [comment]
Let me know if that looks alright. I've added it in both places.

File: tests/file_storage/tests.py [hunk 4]
```diff
@@ -403,6 +403,19 @@ def test_delete_no_name(self):
         with self.assertRaises(AssertionError):
             self.storage.delete('')
 
+    def test_seekable(self):
```
Comment (timgraham) [inline tests/file_storage/tests.py]: Can we put this test in `tests/files`? I don't think it's particular to file storage. Also, it seems we should check the result of `file.seekable()`, not just verify the attribute exists. I'd just skip the test on Python 2.

File: docs/ref/files/file.txt [hunk 5]
```diff
@@ -91,6 +91,10 @@ The ``File`` Class
     ``read``, ``readinto``, ``readlines``, ``seek``, ``softspace``, ``tell``,
     ``truncate``, ``writelines``, ``xreadlines``.
 
+        .. versionchanged:: 1.9
```
Comment (MoritzS) [inline docs/ref/files/file.txt]: No need for the indentation here.

Comment (timgraham) [inline docs/ref/files/file.txt]: deindent this by 4 spaces
Also the note should be "self-contained" as described in https://docs.djangoproject.com/en/1.8/internals/contributing/writing-documentation/#documenting-new-features

File: django/core/files/utils.py [hunk 6]
```diff
@@ -24,6 +27,8 @@ def __init__(self, file):
     write = property(lambda self: self.file.write)
     writelines = property(lambda self: self.file.writelines)
     xreadlines = property(lambda self: self.file.xreadlines)
+    if six.PY3:
```
Comment (timgraham) [inline django/core/files/utils.py]: do we need six.PY3? I assume AttributeError will be raised on Python 2 either way.

File: docs/releases/1.9.txt [hunk 7]
```diff
@@ -200,6 +200,10 @@ File Storage
   <django.core.files.storage.Storage.get_valid_name>` is now called when
   the :attr:`~django.db.models.FileField.upload_to` is a callable.
 
+* :meth:`File.seekable()
+  <django.core.files.base.File.seekable>' is now available when
```
Comment (MoritzS) [inline docs/releases/1.9.txt]: should be `` ` `` instead of `'`

File: django/core/files/utils.py (hunk 6 above)
Comment (carsongee) [inline django/core/files/utils.py]: That is how it was originally written, but we decided explicitness was better. I don't have a preference here myself.

**Comment by timgraham** [comment]
buildbot, add to whitelist.

File: docs/releases/1.9.txt [hunk 8]
```diff
@@ -200,6 +200,9 @@ File Storage
   <django.core.files.storage.Storage.get_valid_name>` is now called when
   the :attr:`~django.db.models.FileField.upload_to` is a callable.
 
+* :class:`~django.core.files.File` now has the ``seekable``
+  method now available when running in Python 3.
```
Comment (MoritzS) [inline docs/releases/1.9.txt]: Duplicate "now".

Comment (carsongee) [inline docs/releases/1.9.txt]: Thanks, that should be sorted.

File: docs/releases/1.9.txt [hunk 9]
```diff
@@ -1,4 +1,4 @@
-============================================
```
Comment (MoritzS) [inline docs/releases/1.9.txt]: You removed a line here.

Comment (carsongee) [inline docs/releases/1.9.txt]: :blush:   I've really been on my game for this PR. Sorry about all these little issues.

**Comment by timgraham** [comment]
merged in e93e0c03b28137e26f9249e0b7c458b0d667a74e, thanks!

**Pull request closed without merging**